areas/mysql/src/AzureMcp.MySql/Commands/BaseMySqlCommand.cs

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.CommandLine;
+using System.CommandLine.Parsing;
+using System.Diagnostics.CodeAnalysis;
+using AzureMcp.Core.Commands;
+using AzureMcp.Core.Commands.Subscription;
+using AzureMcp.MySql.Options;
+using Microsoft.Extensions.Logging;
+
+namespace AzureMcp.MySql.Commands;
+
+public abstract class BaseMySqlCommand<
+    [DynamicallyAccessedMembers(TrimAnnotations.CommandAnnotations)] TOptions>(ILogger<BaseMySqlCommand<TOptions>> logger)
+    : SubscriptionCommand<TOptions> where TOptions : BaseMySqlOptions, new()
+{
+    protected readonly Option<string> _userOption = MySqlOptionDefinitions.User;
+
+    protected readonly ILogger<BaseMySqlCommand<TOptions>> _logger = logger;
+
+    protected override void RegisterOptions(Command command)
+    {
+        base.RegisterOptions(command);
+        command.AddOption(_resourceGroupOption);
```

Comment:
This PR #961 will change this.  I would wait for 961 to be merged and then you take that approach.